config/routes.rb

```diff
@@ -0,0 +1,23 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+
+  # HTTP verb = GET, url path = /tasks, controller = TasksController (ex: tasks), action = method within controller (ex: index), name/alias = "tasks"
+  get "/tasks", to: "tasks#index", as: "tasks"
+  get "/tasks/new", to: "tasks#new", as: "add"
+  post "/tasks", to: "tasks#create"
+
+
+  get "/tasks/:id", to: "tasks#show", as: "task"
+  get "/tasks/:id/edit", to: "tasks#edit", as: "edit"
+  patch "/tasks/:id", to: "tasks#complete?", as: "complete_task"
+  patch "/tasks/:id", to: "tasks#update"
```

Comment:
You've got the same verb/URL combo twice here! Because the router reads this file top-to-bottom looking for a match, this means the bottom route will never be used, and no request will be routed to the `update` action. I believe this is what's causing your update to not work.

To fix this, you would need to give the `complete` route a different URL - maybe `/tasks/:id/mark_complete`.